pkg/cloud/services/networking/network.go

```diff
@@ -101,10 +106,13 @@ func (s *Service) ReconcileExternalNetwork(openStackCluster *infrav1.OpenStackCl
 
 	switch len(networkList) {
 	case 0:
-		// Not finding an external network is fine
-		openStackCluster.Status.ExternalNetwork = &infrav1.NetworkStatus{}
-		s.scope.Logger().Info("No external network found - proceeding with internal network only")
-		return nil
+		if iTrue {
```

Comment:
Neat trick. We could maybe name the var `isAutodetecting` or something to make it simpler to understand.